File: #349 Intersection of Two Arrays/349.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        nums1.sort()
        nums2.sort()
        ans = []
        i=0
        j=0
        print nums1
        print nums2
        while i<len(nums1) and j<len(nums2):
            if nums1[i] < nums2[j]: i+=1
            elif nums1[i] > nums2[j]: j+=1
            else:
                if len(ans) == 0 or nums1[i] != ans[-1]:
                    ans.append(nums1[i])
                i+=1
                j+=1
        return ans

File: 349-Intersection of Two Arrays/349.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        int i=0, j=0;
        vector<int> ans;
        std::sort(nums1.begin(), nums1.end()), std::sort(nums2.begin(), nums2.end());
        while (i<nums1.size() && j<nums2.size()){
            if (nums1[i] < nums2[j]) ++i;
            else if (nums1[i] > nums2[j]) ++j;
            else {
                if (ans.size() == 0 || nums1[i] != ans.back())
                    ans.push_back(nums1[i]);
                ++i, ++j;
            }
        }
        return ans;
    }
};